### PR TITLE
Upgrade thanos version

### PR DESCRIPTION
**Pull request by ilangofman**
Signed-off-by: ilangofman <igofman99@gmail.com>

**What this PR does**:

Upgrades the current version of Thanos to the latest main branch. New version: thanos v0.19.1-0.20210729154440-aa148f8fdb28. 

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by ilangofman** [comment]
In the community call, it was mentioned that the Thanos metrics are mapped in Cortex. I wasn't sure where to look for any changes, can someone please point me to it? I've left it in drafts status until I can verify that if there are any new metrics, then they have been mapped.

**Review by pracucci — APPROVED** [review]
This PR LGTM. I would add a CHANGELOG entry for the file permissions change, something like this:

```
* [CHANGE] The directory and file permissions for TSDB blocks stored on disk in ingester, compactor and store-gateway have been restricted to be writeable only by the owner and readable by owner and group. #4394
```

> In the community call, it was mentioned that the Thanos metrics are mapped in Cortex. I wasn't sure where to look for any changes, can someone please point me to it? I've left it in drafts status until I can verify that if there are any new metrics, then they have been mapped.

From a quick look, I think no metrics have been changed, so nothing to do.

**Comment by ilangofman** [comment]
Thanks for taking a look @pracucci. Really appreciate it. I have added the change log entry.

**Review by pstibrany — APPROVED** [review]
changelog entry is not quite correct (blocks on ingester are produced by prometheus code, which isn't changed, files generated by thanos code aren't group readable), but otherwise lgtm.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 * [CHANGE] Memberlist: the `memberlist_kv_store_value_bytes` has been removed due to values no longer being stored in-memory as encoded bytes. #4345
 * [CHANGE] Prevent path traversal attack from users able to control the HTTP header `X-Scope-OrgID`. #4375 (CVE-2021-36157)
   * Users only have control of the HTTP header when Cortex is not frontend by an auth proxy validating the tenant IDs
+* [CHANGE] The directory and file permissions for TSDB blocks stored on disk in ingester, compactor and store-gateway have been restricted to be writeable only by the owner and readable by owner and group. #4394
```
Comment (pracucci) [inline CHANGELOG.md:11]: Addresses Peter's comment:

```suggestion
* [CHANGE] Some files and directories created by Mimir components on local disk now have stricter permissions, and are only readable by owner, but not group or others. #4394
```

**Pull request merged**